Mathlib/CategoryTheory/Limits/Shapes/Biproducts.lean

```diff
@@ -1239,6 +1256,20 @@ def functoriality (P Q : C) (F : C ⥤ D) [Functor.PreservesZeroMorphisms F] :
       winl := by simp [-BinaryBiconeMorphism.winl, ← f.winl]
       winr := by simp [-BinaryBiconeMorphism.winr, ← f.winr] }
 
+instance functorialityFull [Full F] [Faithful F] : Full (functoriality P Q F) where
+  preimage t :=
+    { hom := F.preimage t.hom
+      winl := F.map_injective (by simpa using t.winl)
+      winr := F.map_injective (by simpa using t.winr)
+      wfst := F.map_injective (by simpa using t.wfst)
+      wsnd := F.map_injective (by simpa using t.wsnd) }
+
+instance functorialityFaithful [Faithful F] : Faithful (functoriality P Q F) where
+  map_injective {c} {c'} f g e := by
+    apply BinaryBiconeMorphism.ext f g
+    let f := BinaryBiconeMorphism.mk.inj e
+    apply F.map_injective f
```

Comment:
```suggestion
    ext
    exact F.map_injective (congr_arg BinaryBiconeMorphism.hom e)
```

Reply:
I ended up using a fully term-mode golf instead, and applied it to the other two instances already in mathlib.